src/math.F90

```diff
@@ -702,6 +718,95 @@ function watt_spectrum(a, b) result(E_out)
 
   end function watt_spectrum
 
+!===============================================================================
+! FADDEEVA the Faddeeva function, using Stephen Johnson's implementation
+!===============================================================================
+
+  function faddeeva(z) result(wv)
+    complex(C_DOUBLE_COMPLEX), intent(in) :: z ! The point to evaluate Z at
+    complex(8)     :: wv     ! The resulting w(z) value
+    real(C_DOUBLE) :: relerr ! Target relative error in inner loop of MIT
+                             !  Faddeeva
+
+    ! Technically, the value we want is given by the equation:
+    ! w(z) = I/Pi * Integrate[Exp[-t^2]/(z-t), {t, -Infinity, Infinity}]
+    ! as shown in Equation 63 from Hwang, R. N. "A rigorous pole
+    ! representation of multilevel cross sections and its practical
+    ! applications." Nuclear Science and Engineering 96.3 (1987): 192-209.
+    !
+    ! The MIT Faddeeva function evaluates w(z) = exp(-z^2)erfc(-iz). These
+    ! two forms of the Faddeeva function are related by a transformation.
+    !
+    ! If we call the integral form w_int, and the function form w_fun:
+    ! For imag(z) > 0, w_int(z) = w_fun(z)
+    ! For imag(z) < 0, w_int(z) = -conjg(w_fun(conjg(z)))
+
+    relerr = ZERO
+    if (aimag(z) > ZERO) then
+      wv = faddeeva_w(z, relerr)
```

Comment:
So, the cost of MIT W is 50% higher than the WHOPPER W when just evaluated on a long array of Z values.  MIT W costs 12% of a cross section lookup.  As such, at best, we'd see a 4% speed increase switching.  In reality, it would be far closer to 1 or 2%, and last time I actually tested it, there was no measurable difference in speed.

Choice of Fadeeva has a moderate impact on accuracy. In the absolute worst case, WHOPPER W would introduce 10^-3 relerr on XS.  QuickW was nearer to 10^-2. 

As I could not find an algorithm that was comfortably 10^-5 everywhere while being faster than MIT W, I stuck with it.